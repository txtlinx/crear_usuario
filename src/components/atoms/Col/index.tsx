import React, { ReactNode } from 'react';
import styled from 'styled-components';

type ColProps = {
  backgroundColor: string;
  textAlign: string;
  sm: number;
  md: number;
  xl: number;
  children: ReactNode;
};

const StyledCol = styled.div<ColProps>`
  text-align: ${(props) => props.textAlign};
  background-color: ${(props) => props.backgroundColor};
  grid-column: span 12;

  @media (min-width: 320px) {
    grid-column: ${(props) => `span ${props.sm}`};
  }
  @media (min-width: 481px) {
    grid-column: ${(props) => `span ${props.sm}`};
  }
  @media (min-width: 641px) {
    grid-column: ${(props) => `span ${props.md}`};
  }
  @media (min-width: 961px) {
    grid-column: ${(props) => `span ${props.md}`};
  }
  @media (min-width: 1025px) {
    grid-column: ${(props) => `span ${props.xl}`};
  }
  @media (min-width: 1281px) {
    grid-column: ${(props) => `span ${props.xl}`};
  }
`;

const Col = ({
  backgroundColor,
  children,
  textAlign,
  sm,
  md,
  xl,
}: ColProps) => {
  return (
    <StyledCol
      backgroundColor={backgroundColor}
      sm={sm}
      md={md}
      xl={xl}
      textAlign={textAlign}
    >
      {children}
    </StyledCol>
  );
};

Col.defaultProps = {
  backgroundColor: 'white',
  textAlign: 'left',
  sm: 12,
  md: 12,
  xl: 12,
};

export default Col;
