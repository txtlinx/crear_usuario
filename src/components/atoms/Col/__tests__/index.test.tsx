import React from 'react';
import { render, screen } from '@testing-library/react';
import Col from '..';

const children = 'Test children';

test('render Col with mandatory props', () => {
  const { container } = render(<Col children={children} />);
  expect(container).toMatchSnapshot();
  const linkElement = screen.getByText(/Test children/i);
  expect(linkElement).toBeInTheDocument();
});
