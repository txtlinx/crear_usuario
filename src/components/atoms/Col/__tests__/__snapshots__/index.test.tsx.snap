// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Col with mandatory props 1`] = `
<div>
  <div
    class="sc-bcXHqe dqsNFd"
  >
    Test children
  </div>
</div>
`;
